client/state/index.js

```diff
@@ -17,11 +18,12 @@ import ui from './ui/reducers';
 const reducer = combineReducers( {
 	sharing,
 	sites,
-	ui
+	ui,
+    notices
 } );
 
-export function createReduxStore() {
-	return applyMiddleware(
-		thunkMiddleware
-	)( createStore )( reducer );
-};
+const store = applyMiddleware(
+	thunkMiddleware
+)( createStore )( reducer );
+
+export default store;
```

Comment:
Since server-side rendering is the critical piece that justifies instantiating the store on-the-fly, we should be clear about our needs there.

>What sense does keeping an application state make for SSR? How do we match state of store with the url to serve a page with a different state?

Because our goal is to serve the same markup from the server as would be rendered in the final state of the client rendering, the server must provide an instance of a Redux store to be used by the component hierarchy in retrieving data. The worry about "leaking data" is that, since a Node server is long-lived, saving a reference to a singleton instance of a store may result in data being shared between requests. If one user's request to the server causes private information to be added to the store and that store is a singleton instance, another user's request may inadvertently gain access to that information. By creating a new store for each and every request, we ensure that this can never happen.

There are quite a few steps between now and being able to render the same on the server as we do on the client, notably:
- Adopt a library that allows us to create instanced stores (Redux)
- Migrate to a global `<App />` component so that we're only rendering a single React tree during the lifecycle of a page
- Reconstruct the route handlers for use in the context of a server, since `page.js` only works in the browser
- Create a mechanism to describe the data needs for a page or set of components, such that the server can detect when the response is ready to be served (after all data has finished fetching)
- Transmit data from the server to the client to avoid unnecessary/duplicated fetching. Redux makes this quite easy by simply making the value of `getState` available as a global variable on the server-side, then "rehydrating" on the client by passing the initial state as the second parameter when creating the Redux store instance

So while we have many examples of singleton stores in the codebase currently, these will need to be changed as we move closer to achieving these goals.

>a) use store reference from `context`, but we need to pass it down from controller to component that wants to send a notice. Sometimes, this can be deep.

This is not as difficult as it seems when using [React `context`](https://facebook.github.io/react/docs/context.html) (different from [`page.js` `context`](https://visionmedia.github.io/page.js/#context) in your comment). We'll recommend using [`react-redux`](https://github.com/rackt/react-redux) to make it even easier. See https://github.com/Automattic/wp-calypso/pull/751#discussion_r47092300 as an example.

>Not every single one of this 84 files is a component, but I'm guessing that would mean connecting most of the components in the whole of calypso. __I can do that, but do we want to reduxify it so much?__

The latter part of this comment would seem to imply that it's difficult or inconvenient to use the Redux store instance, in which case this would be a problem to solve at a framework level and not specifically for notices. Personally, I'd agree that this is not as easy as it was previously when we could simply `import` the module and start using it. Given the considerations above and the conclusion that this is how it must be, we should think about how to ease the transition for those involved in the project.